Source/MythicaEditor/Private/MythicaEditorSubsystem.cpp

```diff
@@ -243,9 +243,11 @@ void UMythicaEditorSubsystem::SetJobsCachedAssetData(int InRequestId, FAssetData
 {
     FMythicaJob* Job = Jobs.Find(InRequestId);
 
-    if (Job)
+    if (Job && Job->CreatedMeshData != InAssetData)
     {
         Job->CreatedMeshData = InAssetData;
+
+        OnGenAssetCreated.Broadcast(InRequestId);
```

Comment:
Makes me think if we moved the logic from UMythicaComponent::UpdateMesh where it scans the import directory to choose the static mesh asset and moved that into UMythicaEditorSubsystem::OnResultMeshData before it marks the state as completed, it may make this simpler to avoid the extra broadcast?

Reply:
It is also a bit awkward because UMythicaComponent::UpdateMesh may call SetJobsCachedAssetData multiple times. But I guess that makes sense that you just need to choose one of them to be displayed in the UI thumbnail.

Reply:
These are some good comments but I was trying not to focus on the architecture too much until I am able to split up MythicaEditorSubsystem into smaller chunks.